gio/tests/vfs.rs

```diff
@@ -0,0 +1,138 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+//
+// The following tests rely on a custom type `MyLocalVfs` that extends the existing GIO type `GLocalVfs`.
+// For each virtual method defined in class `gio::ffi::GVfsClass`, a test checks that `MyLocalVfs` and `GLocalVfs` return the same results.
+// Note that a `MyLocalVfs` instance is built explicitly by calling `glib::Object::builder` whereas a a `GLocalVfs` instance is created by calling `gio::auto::Vfs::local`.
+
+use gio::{prelude::*, subclass::prelude::*, File, Vfs};
+use glib::translate::ToGlibPtr;
+
+// Binding of existing GIO type GLocalVfs.
+mod ffi {
+    use gio::ffi;
+
+    #[derive(Copy, Clone)]
+    #[repr(C)]
+    pub struct GLocalVfs {
+        pub parent_instance: ffi::GVfs,
+    }
+
+    #[derive(Copy, Clone)]
+    #[repr(C)]
+    pub struct GLocalVfsClass {
+        pub parent_class: ffi::GVfsClass,
+    }
+}
+
+glib::wrapper! {
+    #[doc(alias = "GLocalVfs")]
+    pub struct LocalVfs(Object<ffi::GLocalVfs, ffi::GLocalVfsClass>) @extends Vfs;
+
+    match fn {
+        type_ => || {
+            use std::sync::Once;
+            static ONCE: Once = Once::new();
+
+            // ensure type is initialized by calling `gio::auto::File::for_path` to create a `GLocalFile` instance.
+            ONCE.call_once(|| unsafe {
+                let _ = File::for_path("path");
+            });
+            glib::gobject_ffi::g_type_from_name("GLocalVfs".to_glib_none().0)
```

Comment:
Why? You could use the `ObjectSubclass` variant of the `glib::wrapper!` macro here

Reply:
Oh, nevermind